how-to-run-a-node/how-to-run-a-thor-solo-node.md

```diff
@@ -0,0 +1,149 @@
+
+A [Thor] (https://github.com/vechain/thor) Solo Node is a powerful tool for developers, offering a sandbox environment to interact with the VeChainThor blockchain. This guide will walk you through the setup and operation of your own Thor Solo Node, enabling you to test and develop applications in a controlled setting.
+
+## Prerequisites
+
+Before we begin, ensure you have:
+
+* `Go` 1.17+ installed
+* A `C` compiler
+* Git
+
+## Installation Process
+
+Clone the Thor repository:
+
+```bash
+git clone https://github.com/vechain/thor
+```
+
+Navigate to the Thor directory and build:
+
+```bash
+cd thor
+make
+```
+
+Upon successful compilation, you'll find the `thor` binary in the `bin` directory.
+
+## Key Command-line Options
+
+Explore Thor's capabilities with `./bin/thor -h`. Essential options include:
+
+* `--api-cors '*'`: Accept cross-origin requests from any domain.
+* `--api-addr value`: Set API service listening address (default: "localhost:8669").
+* `--api-call-gas-limit value`: Limit contract call gas (default: 50000000).
+* `--verbosity value` log verbosity (0-9) (default: 3).
+
+## Advanced Configuration
+
+### Sub-Commands for Enhanced Control
+
+Thor offers versatile sub-commands:
+
+```bash
+./bin/thor solo --on-demand            # Create new blocks for pending transactions
+./bin/thor solo --persist              # Save blockchain data to disk
+./bin/thor solo --persist --on-demand  # The two options can work together
+```
+
+### Enabling Remote Access
+
+If you are not running Thor on the same host as the development environment then you need to provide an API listening address using the `--api-addr` command-line option. For example to make Thor accept any remote connection:
+
+```bash
+./bin/thor solo --on-demand --api-addr 0.0.0.0:8669
+```
+
+### Debugging with Increased Verbosity
+
+The default verbosity option in Thor (3) might not be providing enough information for debugging. Using the `--verbosity` command-line option, we can increase the amount of information Thor prints in stdout. For example:
+
+<pre class="language-bash"><code class="lang-bash"><strong>./bin/thor solo --on-demand --verbosity 4
+</strong></code></pre>
+
+### Master Key Management
+
+Secure your node with master key commands:
+
+```bash
+# View master address
+./bin/thor master-key
+
+# Export master key to keystore
+./bin/thor master-key --export > keystore.json
+
+# Import master key from keystore
+cat keystore.json | ./bin/thor master-key --import
+```
+
+## RESTful API: A Developer's Playground
+
+Thor's RESTful API offers a user-friendly interface for blockchain interaction. Access the Stoplight UI at:
+
+```bash
+http://127.0.0.1:8669/doc/stoplight-ui/
+```
+
+Or the Swagger UI at:
+
+```bash
+http://127.0.0.1:8669/doc/swagger-ui/
+```
+
+If Thor is running on a different host, make sure you use the IP of said host as well as the `--api-addr` command-line option when running Thor.
```

Comment:
```suggestion
If Thor is running on a different host, make sure to run it using the IP of said host instead of the localhost, as well as the `--api-addr` command-line option.
```